allennlp/semparse/domain_languages/domain_language.py

```diff
@@ -361,15 +400,31 @@ def action_sequence_to_logical_form(self, action_sequence: List[str]) -> str:
             raise ParsingError("Extra actions in action sequence")
         return nltk_tree_to_logical_form(tree)
 
-    def add_predicate(self, name: str, function: Callable):
+    def add_predicate(self, name: str, function: Callable, side_arguments: List[str] = None):
         """
         Adds a predicate to this domain language.  Typically you do this with the ``@predicate``
         decorator on the methods in your class.  But, if you need to for whatever reason, you can
         also call this function yourself with a (type-annotated) function to add it to your
         language.
+
+        Parameters
+        ----------
+        name : ``str``
+            The name that we will use in the induced language for this function.
+        function : ``Callable``
+            The function that gets called when executing a predicate with the given name.
+        side_arguments : ``List[str]``, optional
+            If given, we will ignore these arguments for the purposes of grammar induction.  This
+            is to allow passing extra arguments from the decoder state that are not explicitly part
+            of the language the decoder produces, such as the decoder's attention over the question
+            when a terminal was predicted.  If you use this functionality, you also need to
+            pass these arguments when you call ``language.execute(logical_form)``.  See
```

Comment:
You mean `language.execute_action_sequence`, right? Also in the line below.

Reply:
Thanks, I decided I needed to split that out into a new function after I wrote this docstring.  Fixed.